lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -0,0 +1,344 @@
+  function handleComponentFocus(evt: FocusEvent<HTMLDivElement>) {
+    // check if this is focus coming into the component from somewhere else on the page, not just moving between
+    // children of this component and not from focus coming into the browser from some other window
+    const comingFromOutsidePage = evt.relatedTarget === null,
+        comingFromChildNode = evt.relatedTarget instanceof Node && evt.currentTarget.contains(evt.relatedTarget);
+
+    if (!(comingFromOutsidePage || comingFromChildNode) && error) {
+      doSearch(searchText);
+    }
+    else if (error) {
+      setShowDropdown(true);
+    }
+  }
+
+  // In the blur event, we use the relatedTarget to check if the focused element is a child of the parent
+  // to show/hide dropdown, however, clicking the button in Safari does not focus it, which means the
+  // relatedTarget will be null, this is to stop the mouse down event being fired for the element that
+  // is listening to the blur event, in this case, the show/hide dropdown is handled by the click event
+  function handleMouseDown(evt: MouseEvent) {evt.preventDefault(); }
+
+  function handleComponentBlur(evt: FocusEvent<HTMLDivElement>) {
+    setElToFocusId('');
+    setFocusableBtnIndex(null);
+
+    // Check if the new focused element is a child of the parent, if not, then close the dropdown menu
+    if (!(evt.relatedTarget instanceof Node && evt.currentTarget.contains(evt.relatedTarget))) {
+      setShowDropdown(false);
+    }
+  }
+
+  function handleFilterChange(value: string) {
+    setElToFocusId('');
+    setFocusableBtnIndex(null);
+    onSearchTextChange(value);
+
+    if (value && value.trim() !== searchText.trim()) {
+      doSearch(value);
+      setPrevSearchText('');
+    }
+    else if (!value) {
+      setShowDropdown(false);
+    }
+  }
+
+  // helper for focusing different buttons in the dropdown menu
+  const adjustBtnFocus = (adjust: (i: number) => number) => () => {
+        const newFocusableBtnIndex = adjust(focusableBtnIndex ?? 0),
+            elToFocus = dropdownRef.current?.children[newFocusableBtnIndex] as HTMLElement | null;
+        if (elToFocus) {
+          setElToFocusId(elToFocus.id);
+          setFocusableBtnIndex(newFocusableBtnIndex);
+          if (autoComplete && typeof elToFocus.textContent === 'string') {
+            onSearchTextChange(elToFocus.textContent);
+          }
+        }
+      },
+      focusNext = adjustBtnFocus(inc),
+      focusPrev = adjustBtnFocus(dec),
+      focusFirst = adjustBtnFocus(always(0)),
+      focusLast = adjustBtnFocus(always(matches.length - 1));
+
+  function handleKeyDown(evt: KeyboardEvent<HTMLElement>) {
+    const inputEle = evt.currentTarget as HTMLInputElement;
+    switch (evt.key) {
+      case 'Enter':
+        if (typeof focusableBtnIndex === 'number' && elToFocusId) {
+          handleOnClick(matches[focusableBtnIndex]);
+        }
+        evt.preventDefault();
```

Comment:
Why the `preventDefault`? I assume that would prevent the combobox from matching the typical form field behavior of Enter submitting the form. Reading the ARIA Patterns document, it's unclear whether that's desired or not

Reply:
Yes to prevent from submitting the form.